mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -581,15 +605,34 @@ impl MakerSwap {
 
     async fn negotiate(&self) -> Result<(Option<MakerSwapCommand>, Vec<MakerSwapEvent>), String> {
         let negotiation_data = self.get_my_negotiation_data();
+        let mut msgs = vec![];
+
+        // Important to try the new versioned negotiation message first
+        if cfg!(not(feature = "for-tests")) || !env_var_as_bool("USE_NON_VERSIONED_MAKER") {
+            let maker_versioned_negotiation_msg = SwapMsgExt::NegotiationVersioned(NegotiationDataMsgVersion {
+                version: SWAP_PROTOCOL_VERSION,
+                msg: negotiation_data.clone(),
+            });
+            msgs.push((
+                swap_ext_topic(&self.uuid),
+                SwapMsgWrapper::Ext(maker_versioned_negotiation_msg),
+            ));
+        }
+
+        let maker_old_negotiation_msg = SwapMsg::Negotiation(negotiation_data);
+        msgs.push((
+            swap_topic(&self.uuid),
+            SwapMsgWrapper::Legacy(maker_old_negotiation_msg),
+        ));
```

Comment:
could you clarify, are you pushing both versioned negotiation msg and legacy negotiation msg in production code?
could you tell why do we need this?

Reply:
We need this for the upgrade period: first try if the remote node understands the old (default) version, if yes, fallback to older negotiation protocol and turn off new features